packages/next/client/script.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useEffect, useContext } from 'react'
+import { ScriptHTMLAttributes } from 'react'
+import { HeadManagerContext } from '../next-server/lib/head-manager-context'
+
+const ScriptCache = new Set()
+
+const DOMAttributeNames: Record<string, string> = {
+  acceptCharset: 'accept-charset',
+  className: 'class',
+  htmlFor: 'for',
+  httpEquiv: 'http-equiv',
+}
+
+interface Props extends ScriptHTMLAttributes<HTMLScriptElement> {
+  strategy?: 'defer' | 'lazy' | 'dangerousRenderBlocking' | 'eager'
```

Comment:
I'd prefer to only start with the `lazy` option, which I assume is we want to be the default.

Do we really want to give users the ability to do `eager` with the built-in component?

Also, it's not clear to me how `eager` differs from `dangerousRenderBlocking`.

Reply:
We really want to test these options and see how they do in production. Restricting the options would limit the range of the experiment.

eager - Loads script after FCP - Very useful for scripts like polyfills which have to execute before 1P scripts
dangerousRenderBlocking - Usually loads script in place - gives users a fallback to the default loading mechanism